aws/internal/service/ec2/finder/finder.go

```diff
@@ -55,6 +55,19 @@ func ClientVpnRouteByID(conn *ec2.EC2, routeID string) (*ec2.DescribeClientVpnRo
 	return ClientVpnRoute(conn, endpointID, targetSubnetID, destinationCidr)
 }
 
+func InternetGatewayAttachmentByID(conn *ec2.EC2, igwID, vpcID string) (*ec2.DescribeInternetGatewaysOutput, error) {
```

Comment:
I would have this return the actual attachment (or `nil`) and deal with all the

```go
if result == nil || len(result.InternetGateways) == 0 || result.InternetGateways[0] == nil {
}
```

logic here rather than in the `Status` function.

```suggestion
func InternetGatewayAttachment(conn *ec2.EC2, igwID, vpcID string) (*ec2.InternetGatewayAttachment, error) {
```